.github/workflows/run-api-test.yml

```diff
@@ -120,32 +121,67 @@ jobs:
 
           if [ $n -ge $max_retries ]; then
             echo "All $max_retries attempts failed."
-            exit 1
+            exit_code=1
           fi
 
+          echo "last_exit_code=$exit_code" >> "$GITHUB_ENV"
+          exit "$last_exit_code"
+
+      - name: Upload artifact if exit code 53
```

Comment:
See description
```
For the API tests, to assert whether the test actually failed while running our code, I made it throw exit code 53, which is a random code I devised that should not match any other orchestration failure from deno, dependencies or otherwise.
```